core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -198,12 +198,21 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
   }
 
   inline static int scratch_size_max(int level) {
-    return (
-        level == 0 ? 1024 * 40 :  // 48kB is the max for CUDA, but we need some
-                                  // for team_member.reduce etc.
-            20 * 1024 *
-                1024);  // arbitrarily setting this to 20MB, for a Volta V100
-                        // that would give us about 3.2GB for 2 teams per SM
+    // Cuda Teams use (team_size + 2)*sizeof(double) shared memory for team
```

Comment:
I assume the `sizeof(double)` here is our saying "let the value type for the reduction be double"?

Reply:
No, it actually doesn't I had there an 8 before because we actually say sizeof(double) in the code, but if you happen to do reductions with value_types larger than that they are done in stages.

Reply:
But I think its not a great idea to write the entire explanation of how our reductions work in the description of one variable.

Reply:
OK